docs/overview/hooks.md

```diff
@@ -14,60 +14,84 @@ title: Lifecycle hooks overview
     <a style="font-style:italic;padding:5px;margin:5px;"  href="https://egghead.io/lessons/react-loading-data-from-the-server">Hosted on egghead.io</a>
 </details>
 
-All of the below hooks can be created by returning an action with the given name, like:
+`mobx-state-tree` supports passing a variety of hooks that are called throughout a node's lifecycle. Hooks are passes as actions with the name of the hook, like:
 
 ```javascript
-const Todo = types.model("Todo", { done: true }).actions(self => ({
-    afterCreate() {
-        console.log("Created a new todo!")
-    }
+const Todo = types.model("Todo", { done: true }).actions((self) => ({
+  afterCreate() {
+    console.log("Created a new todo!")
+  }
 }))
 ```
 
-_⚠ The section below is outdated, and should be updated to use [`types.snapshotProcessor`](/API/#snapshotprocessor) instead of the snapshot hooks⚠_
-
-The exception to this rule are the `preProcessSnapshot` and `postProcessSnapshot` hooks (see `types.snapshotProcessor` as an alternative):
-
-```javascript
-types
-    .model("Todo", { done: true })
-    .preProcessSnapshot(snapshot => ({
-        // auto convert strings to booleans as part of preprocessing
-        done: snapshot.done === "true" ? true : snapshot.done === "false" ? false : snapshot.done
-    }))
-    .actions(self => ({
-        afterCreate() {
-            console.log("Created a new todo!")
-        }
-    }))
-```
-
 Note: pre and post processing are just meant to convert your data into types that are more acceptable to MST. Typically it should be the case that `postProcess(preProcess(snapshot)) === snapshot. If that isn't the case, consider whether you shouldn't be using a dedicated a view instead to normalize your snapshot to some other format you need.
```

Comment:
**suggestion:** @airhorns - we should remove or modify this since it would no longer be strictly true with your changes.

Reply:
Updated this -- moved it down closer to the snapshot section as well